src/include/utime.h

```diff
@@ -32,6 +32,10 @@
 // --------
 // utime_t
 
+//from __u64 to __u32
+inline __u32 ut_safe_convert(__u64 t) {
+  return t < std::numeric_limits<uint32_t>::max() ? t : std::numeric_limits<uint32_t>::max();
```

Comment:
i'd suggest rename `ut_safe_convert()` to `cap_to_u32_max()`. also might want to `use std::min()` here.